### PR TITLE
feat: add main logic of authz

### LINKED ISSUE #423: Provide an official middleware/solution/starterkit for GraphQL based on Casbin
Some people want this in another repo:
https://github.com/graphql-nexus/nexus/issues/238
https://github.com/99designs/gqlgen/issues/937

Some people asked us:
https://github.com/casbin/casbin/issues/327

Some people even made some demo solution:
https://github.com/esmaeilpour/graphql-casbin

Some companies like Auth0 also have solution:
https://auth0.com/blog/authorization-series-pt-3-dynamic-authorization-with-graphql-and-rules/

A reddit post that asks for Casbin support on GraphQL: https://www.reddit.com/r/golang/comments/i9drqp/graphql_and_authorization/

I think we should make an official solution for how to use Casbin in GraphQL. Can anyone work on this?

### PR DESCRIPTION
**Pull request by kilosonc**
Signed-off-by: closetool <c299999999@qq.com>

Fix: casbin/casbin#423

### CODE REVIEW

**Comment by kilosonc** [comment]
@hsluoyz plz review

File: README.md [hunk 1]
```diff
@@ -1 +1,23 @@
...
+
+Enforce Example:
+
+```go
+e, _ := casbin.NewEnforcer("./examples/model.conf", "./examples/policy.csv")
+schema := authz.InitType(e)
+result := authz.Execute(`{enforce(sub:"alice" obj:"data1" act:"read"){sub obj act ok}}`, schema)
+```
+
+More info, please refer to [graphql](https://github.com/graphql/graphql-js) and [basic usage](./authz_test.go)
```
Comment (hsluoyz) [inline README.md:23]: Is graphql-js for Go or JS?

File: README.md [hunk 2]
```diff
@@ -1 +1,23 @@
...
+```
+
+## Usage
+
+Enforce Example:
+
+```go
+e, _ := casbin.NewEnforcer("./examples/model.conf", "./examples/policy.csv")
+schema := authz.InitType(e)
+result := authz.Execute(`{enforce(sub:"alice" obj:"data1" act:"read"){sub obj act ok}}`, schema)
```
Comment (hsluoyz) [inline README.md:20]: How to let this repo act as a middleware of graphql? Does graphql provide a middleware mechanism?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (kilosonc) [reply README.md:23]: For graphql is origined from js

File: README.md (hunk 2 above)
Comment (kilosonc) [reply README.md:20]: If we need to make a authz middleware, we could just using `Enforce` method, why graphql is needed?
Graphql is something like restful api, I don't think it should be used as middleware.
Maybe a http service will be more suitable, like https://github.com/closetool/graphql-authz/blob/51cffa025cfe733a4fd2ae6f1e512425c9dfccff/authz_test.go#L46

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**